plugins/catalog/src/alpha.tsx

```diff
@@ -65,12 +81,203 @@ export const CatalogSearchResultListItemExtension =
       ),
   });
 
+/** @alpha */
+export function createCatalogFilterExtension<
+  TInputs extends AnyExtensionInputMap,
+  TConfig = never,
+>(options: {
+  id: string;
+  inputs?: TInputs;
+  configSchema?: PortableSchema<TConfig>;
+  loader: (options: { config: TConfig }) => Promise<JSX.Element>;
+}) {
+  return createExtension({
+    id: `catalog.filter.${options.id}`,
+    at: 'catalog/filters',
+    inputs: options.inputs ?? {},
+    configSchema: options.configSchema,
+    output: {
+      element: coreExtensionData.reactElement,
```

Comment:
I feel we need a different type here, since you can't really install any plain react element here and expect it to work. Maybe a just a plain table filter type that also forwards a react element, although a more elaborate refactor could be to investigate if it's possible to break the contract apart a bit more similar to scaffolder field extensions, but I am worried that'd get very messy very quickly.

Reply:
Could you please point me to the ScaffolderFields implementation file?

Reply:
This kind of things: https://github.com/backstage/backstage/blob/cbed529916fa87ac0f1145d5f9a1340170dfb838/plugins/scaffolder-react/src/extensions/index.tsx#L39. The key there being that it's an element as well as additional configuration surrounding it. Tbh it's what we're already doing a lot in DI.

Thing is that we currently have a contract for how the field interacts with the outside through the react context that the parent page provides. This is a quite hidden interface and completely opaque to the core framework. If we instead lifted out the context and refactored it into output that the filter extensions provide then we'd just have a single place where the communication happens and a more explicit connection between the parent page and the filter extensions.

I don't think any of that is worth doing at this point thought and it may not be the best design either. Happy to leave this as a known thing and see where else we encounter communication through react context and at a broader scale decide what we want to do with that.

Reply:
Thanks for sharing, I agree with you, let's leave it as is and refactor later when we have more use cases 🙂